### PR TITLE
DashboardScenes: Fix panel links to update on variable or timerange change

### PR DESCRIPTION
**Pull request by mdvictor**
**What is this feature?**

Fixes an issue with scenes dashboards where the panel links would not update URLs that contained variable or timerange data when either of those changed.

**Why do we need this feature?**

Fix the issue and make it work like in the old arch

**Who is this feature for?**

Everyone

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/features/dashboard-scene/scene/PanelLinks.tsx [hunk 1]
```diff
@@ -11,11 +20,21 @@ interface VizPanelLinksState extends SceneObjectState {
...
+    },
+  });
+
   static Component = VizPanelLinksRenderer;
 }
 
 function VizPanelLinksRenderer({ model }: SceneComponentProps<VizPanelLinks>) {
   const { menu, rawLinks } = model.useState();
+  const dashboard = getDashboardSceneFor(model);
+  sceneGraph.getTimeRange(dashboard).useState();
```
Comment (torkelo) [inline public/app/features/dashboard-scene/scene/PanelLinks.tsx:37]: why not use `sceneGraph.getTimeRange(model).useState()`  ?

Comment (mdvictor) [reply public/app/features/dashboard-scene/scene/PanelLinks.tsx]: didn't think that through, you're right!

**Review by axelavargas — APPROVED** [review]
Works as expected 🥇

**Comment by tolzhabayev** [comment]
Can this be backported to 11.3.x branch? @axelavargas

**Pull request merged**